src/EnergyPlus/CurveManager.cc

```diff
@@ -442,60 +438,62 @@ namespace CurveManager {
                                                      state.dataIPShortCut->cAlphaFieldNames(1),
                                                      ErrorsFound);
             ++CurveNum;
-            state.dataCurveManager->PerfCurve(CurveNum).Name = Alphas(1);
-
-            state.dataCurveManager->PerfCurve(CurveNum).curveType = CurveType::ChillerPartLoadWithLift;
-            state.dataCurveManager->PerfCurve(CurveNum).ObjectType = CurrentModuleObject;
-            state.dataCurveManager->PerfCurve(CurveNum).NumDims = 3;
-            state.dataCurveManager->PerfCurve(CurveNum).InterpolationType = InterpType::EvaluateCurveToLimits;
-
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff1 = Numbers(1);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff2 = Numbers(2);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff3 = Numbers(3);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff4 = Numbers(4);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff5 = Numbers(5);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff6 = Numbers(6);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff7 = Numbers(7);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff8 = Numbers(8);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff9 = Numbers(9);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff10 = Numbers(10);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff11 = Numbers(11);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff12 = Numbers(12);
-
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Min = Numbers(13);
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Max = Numbers(14);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Min = Numbers(15);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Max = Numbers(16);
-            state.dataCurveManager->PerfCurve(CurveNum).Var3Min = Numbers(17);
-            state.dataCurveManager->PerfCurve(CurveNum).Var3Max = Numbers(18);
+            auto &thisCurve = state.dataCurveManager->PerfCurve(CurveNum);
+
+            thisCurve.Name = Alphas(1);
+
+            thisCurve.curveType = CurveType::ChillerPartLoadWithLift;
+            thisCurve.ObjectType = CurrentModuleObject;
+            thisCurve.NumDims = 3;
+            thisCurve.InterpolationType = InterpType::EvaluateCurveToLimits;
+
+            thisCurve.coeff[0] = Numbers(1);
+            thisCurve.coeff[1] = Numbers(2);
+            thisCurve.coeff[2] = Numbers(3);
```

Comment:
Also, wondering if you want to do with `Curve` what @yujiex is currently doing with `Material`, which is create a `CurveBase` class that contains `name`, `objectType`, `curveType`, `numDims`, `interpolationType` and then child classes for each `CurveType` (you know, like inheritance and stuff). Actually, there should be a global `EPObjectBase` class with `name` and `objectType` and `MaterialBase` and `CurveBase` should derive from it.

Reply:
Yep, that's the plan.  I'm trying to make a first phase where that functionality is added inside CurveManager, but hidden from the user through the current GetCurveIndex and CurveValue hanging functions with index arguments.  That way we can then propagate a better pattern through all the client code incrementally just like the SolveRoot work, and not have it be an atomic change that affects 120 files.

Reply:
One thing I forgot to mention here is this will require the Curve array itself to become an array of `CurveBase *` rather than `Curve` objects. We are actually doing that first before the inheritance stuff in `Material` because it makes it easier to do the inheritance incrementally later, i.e., to have one giant `CurveChild` object and then incrementally splitting classes off of it.

Reply:
I definitely have some thoughts on that.  I may propose an alternative from what we've learned with the PlantComponent base class, but either way, yes, a base class should be forming up very soon.

Reply:
I am not sure what you mean by "the thing we did with plant component". However, I am not proposing anything different in terms of memory management (i.e., ownership) or overall reference structure, just: i) "right-sizing" objects so that we are not allocating huge tracts of unused memory and destroying locality, and ii) taking advantage of inheritance using a combination of shared functions and virtual functions as appropriate.